src/arrays/create.rs

```diff
@@ -139,62 +142,79 @@ impl <N: Numeric> ArrayCreate<N> for Array<N> {
             .collect()
     }
 
-    fn logspace_a(start: &Self, stop: &Self, num: Option<usize>, endpoint: Option<bool>, base: Option<&Array<usize>>) -> Self {
+    fn logspace_a(start: &Self, stop: &Self, num: Option<usize>, endpoint: Option<bool>, base: Option<&Array<usize>>) -> Result<Self, ArrayError> {
         let start = if start.len() == 1 { Self::full_like(stop, start[0]) } else { start.clone() };
         let stop = if stop.len() == 1 { Self::full_like(&start, stop[0]) } else { stop.clone() };
-        assert_eq!(start.get_shape(), stop.get_shape());
+
+        if start.get_shape() != stop.get_shape() { return Err(ArrayError::ShapesMustMatch { s1: "start", s2: "stop", }); }
```

Comment:
I will create macro, because there are a lot of parts of code when you use this condition.
not_eq!(a, b, err)